io/net/bonding.py

```diff
@@ -520,6 +509,34 @@ def test_cleanup(self):
             except Exception:
                 self.log.info("backup file not availbale, could not restore file.")
 
+        if self.peer_bond_needed:
+            self.bond_remove("peer")
+            self.session_pubic_ip = Session(
+                                  self.peer_public_ip, user=self.user,
+                                  password=self.password)
+            for ipaddr, interface in zip(self.peer_ipaddr,
+                                         self.peer_interfaces):
+                if self.peer_bond_needed:
+                    self.remotehost = RemoteHost(
+                                    self.peer_public_ip, self.user,
+                                    password=self.password)
+                    peer_networkinterface = NetworkInterface(interface,
+                                                             self.remotehost)
+                    try:
+                        peer_networkinterface.flush_ipaddr()
+                        peer_networkinterface.add_ipaddr(ipaddr, self.netmask)
+                        peer_networkinterface.bring_up()
+                    except Exception:
+                        peer_networkinterface.save(ipaddr, self.netmask)
+            for val in self.peer_interfaces:
+                cmd = "ip link set dev %s down; ip link set dev %s up; sleep 5"\
+                      % (val, val)
+                output = self.session_public_ip.cmd(cmd)
```

Comment:
@TasmiyaNalatwad I feel we have the bring_down and bring_up utility methods.
why cannot we use them here?

Reply:
@Naresh-ibm  This piece of code was already there before, but was having 'ifup and ifdown' commands which are not supported by rhel 9 on-words. To avoid that risk of failure in future results just modified the ifup and ifdown with 'ip link set dev % up and down' commands other piece of code is kept as it is.

Reply:
@TasmiyaNalatwad  would be good having the lib func for such commands as we will be using them in multiple places.

Reply:
@Naresh-ibm  There is already lib for making link down and up. that is bring_down() / bring_up()
which is already present as part of cleanup. using it again would be redundant code. Can i remove this for loop as it is achieving the same purpose.

@abdhaleegit  Requesting your input on the above query.